src/engine/plugins/nearest.cpp

```diff
@@ -26,7 +26,7 @@ Status NearestPlugin::HandleRequest(const api::NearestParameters &params,
 {
     BOOST_ASSERT(params.IsValid());
 
-    if (params.number_of_results > max_results)
+    if (static_cast<std::int64_t>(params.number_of_results) > max_results)
```

Comment:
Just for the fun of it, shouldn't we use a `boost::numeric_cast` for these kinds of checks (even though safe right now) to prevent errors when the casted type changes?